### ISSUE TITLE
Turn off masking

### ISSUE THREAD

**Issue by Aayushktyagi** [IcarusWizard/MAE#9]
Hi @IcarusWizard,

Great work!!

We are trying to correct segmentation label using masked Autoencoder. I tried MAE and its working good. 
Now for inference I want to pass whole image(without being masked) and check if MAE is able to correct certain regions.
Is there a way where I can turn off mask and infer on whole image? 

let me know if my question is unclear.

Thanks

**Comment by IcarusWizard** [comment]
Hi @Aayushktyagi,

I am glad you find this work useful.

I am not sure if I understand your question correctly. From what I understand you still need the decoder to correct the segmentation for you, right? If that is the case, I think you can do this by overwriting the `encoder.shuffle` to `PatchShuffle(0)` which sets the masking rate to 0, i.e. disable the mask. You also need to change this [line](https://github.com/IcarusWizard/MAE/blob/main/model.py#L106) of the decoder to
```python
if backward_indexes.shape[0] - features.shape[0] > 0:
    features = torch.cat([features, self.mask_token.expand(backward_indexes.shape[0] - features.shape[0], features.shape[1], -1)], dim=0)
``` 

Alternatively, modifying `ViT_Classifier` is also possible. It does not have the masking, but only runs the encoder and uses the output `cls_token` as the feature. You will need to change the selection of output features and incorporate the decoder yourself. 

Let me know if you still have problems, and I am glad to help. :)

**Comment by Aayushktyagi** [comment]
Hi @IcarusWizard ,

Thanks for quick response. What I essentially meant was, is there a way to change the mask_ratio at the time of inference? 
For example, I trained the model for mask_ratio = 0.25. However, I want to vary this ratio during inference between 0-1. 
Suppose I want want to check the above model for mask_ratio = 0.1, 0.5, 0.75 and more. Is it possible to do so?

Will look into your answer.

Thanks

**Comment by IcarusWizard** [comment]
I see. `encoder.shuffle = PatchShuffle(ratio)` should do the job.
